pkg/odo/genericclioptions/context.go

```diff
@@ -72,6 +72,42 @@ func checkProjectCreateOrDeleteOnlyOnInvalidNamespace(command *cobra.Command, er
 	}
 }
 
+func checkValidConfig(command *cobra.Command, configFileName string) {
+	_, err1 := config.NewLocalConfigInfo(configFileName, true)
+	allowedFirstChildCommandList := []string{"project", "catalog", "login", "logout"}
+
+	if command.HasParent() {
+		rootCommand := command.Root()
+		// if command is create operation or push operation just allow it
+		if command.Name() == "create" || command.Name() == "push" {
+			return
+		}
+		// Find the first child of the command. At end of this, c will be the first child
+		p := command.Parent()
```

Comment:
extracting this out in a function like maybe `getFirstChildOfCommand()` will make this more understandable. just a suggestion

Reply:
Done